Makefile

```diff
@@ -384,12 +391,53 @@ release-windows-rebuild: bin/releases/git-lfs-windows-assets-$(VERSION).tar.gz
 		); \
 		status="$$?"; [ -n "$$temp" ] && $(RM) -r "$$temp"; exit "$$status"
 
+# release-darwin is a target that builds and signs Darwin (macOS) binaries.  It must
+# be run on a macOS machine with a suitable version of XCode.
+#
+# You may sign with a different certificate by specifying DARWIN_CERT_SHA1.
+.PHONY : release-darwin
+release-darwin: bin/releases/git-lfs-darwin-amd64-$(VERSION).zip
+	for i in $^; do \
+		temp=$$(mktemp -d) && \
+		( \
+			unzip -d "$$temp" $^ && \
+			codesign --keychain $(DARWIN_KEYCHAIN_ID) -s "$(DARWIN_CERT_ID)" --force --timestamp -vvvv --options runtime "$$temp/git-lfs" && \
+			codesign -dvvv "$$temp/git-lfs" && \
+			zip -j $$i "$$temp/git-lfs" && \
+			codesign --keychain $(DARWIN_KEYCHAIN_ID) -s "$(DARWIN_CERT_ID)" --force --timestamp -vvvv --options runtime "$$i" && \
+			codesign -dvvv "$$i" && \
+			jq -e ".notarize.path = \"$$i\" | .apple_id.username = \"$(DARWIN_DEV_USER)\"" script/macos/manifest.json > "$$temp/manifest.json"; \
+			for j in 1 2 3; \
+			do \
+				gon "$$temp/manifest.json" && break; \
+			done; \
+		); \
+		status="$$?"; [ -n "$$temp" ] && $(RM) -r "$$temp"; [ "$$status" -eq 0 ] || exit "$$status"; \
+	done
+
 .PHONY : release-write-certificate
 release-write-certificate:
 	@echo "Writing certificate to $(CERT_FILE)"
 	@echo "$$CERT_CONTENTS" | base64 --decode >"$$CERT_FILE"
 	@printf 'Wrote %d bytes (SHA256 %s) to certificate file\n' $$(wc -c <"$$CERT_FILE") $$(shasum -ba 256 "$$CERT_FILE" | cut -d' ' -f1)
 
+# release-import-certificate imports the given certificate into the macOS
+# keychain "CI".  It is not generally recommended to run this on a user system,
+# since it creates a new keychain and modifies the keychain search path.
```

Comment:
I don't suppose there's any way to have this target's build commands start by checking that they are running as an Action CI-specific user, and pausing for user confirmation if not?

Reply:
I don't think we can do an Actions-specific user since there's no guarantee that won't change, but we can certainly check for an Actions-specific environment variable and fail otherwise.  Thanks for the idea.

Reply:
I've checked for the `CI` environment variable and gently chided the user if it's not set.